src/player.cpp

```diff
@@ -7599,6 +7620,148 @@ void player::hardcoded_effects(effect &it)
                 }
             }
         }
+    } else if ( id == "skin_rot" || id == "skin_rot_recovery") {
+        bool recovery = id == "skin_rot_recovery" ? true : false;
+        float heal_chance = 7200.0;
+        float spread_chance = 3600.0;
+        float health = get_healthy();
+        float health_mod = health / 100.0 + 1.0;
```

Comment:
`health_mod` here can be anything from -1 to +3, as health goes from -200 to 200.
While float division by 0 doesn't cause crashes, it may result in weirdness as you get infinities.